impl/src/fmt/mod.rs

```diff
@@ -133,28 +133,68 @@ impl ToTokens for FmtAttribute {
 }
 
 impl FmtAttribute {
-    /// Returns an [`Iterator`] over bounded [`syn::Type`]s and trait names.
-    fn bounded_types<'a>(
+    /// Checks whether this [`FmtAttribute`] can be replaced with a trivial delegation (calling a formatting trait
+    /// directly instead of interpolation syntax).
+    ///
+    /// Returns [`syn::Ident`] of the delegated trait, if delegation is possible.
+    fn delegatable(&self) -> bool {
+        // `FmtAttribute` is delegatable when:
+        parsing::format(&self.lit.value())
+            .and_then(|(more, p)| {
+                // (1) There is exactly one formatting parameter.
+                more.is_empty().then_some(p)
+            })
+            .filter(|p| {
+                // (2) And the formatting parameter doesn't contain any modifiers.
+                p.spec
+                    .map(|s| {
+                        s.width.is_none() && s.precision.is_none() && s.ty.is_trivial()
```

Comment:
As far as I can tell this is missing a bunch of modifiers that are listed in the std::fmt docs, specifically:
1. Fill
2. Align
3. Sign
4. Alternate

Reply:
@JelteF good catch. I was thinking that we already parse the whole stuff in the parser, but it seems that while parser considers all the grammar correctly (we have tests for that), it does parse only the necessary minimum.

Reply:
Makes sense, I guess with this PR the necessary minimum changes ;)